core/src/test/java/hudson/model/FingerprintCleanupThreadTest.java

```diff
@@ -90,6 +94,32 @@ public void testIOExceptionOnLoad() throws IOException {
         assertTrue("Should have logged IOException.", logOutput.contains("ERROR: Failed to process"));
     }
 
+    @Test
+    public void testBlockingFacetBlocksDeletion() throws IOException {
+        createFolderStructure();
+        TestTaskListener testTaskListener = new TestTaskListener();
+        Fingerprint fp = new TestFingerprint(false);
+        fp.facets.setOwner(Saveable.NOOP);
+        TestFingperprintFacet facet = new TestFingperprintFacet(fp, System.currentTimeMillis(), true);
+        fp.facets.add(facet);
+        FingerprintCleanupThread cleanupThread = new TestFingerprintCleanupThread(fp);
+        cleanupThread.execute(testTaskListener);
+        assertTrue("Should have deleted obsolete file.", fpFile.toFile().exists());
+    }
+
+    @Test
+    public void testUnblockedFacetsDontBlockDeletion() throws IOException {
+        createFolderStructure();
+        TestTaskListener testTaskListener = new TestTaskListener();
+        Fingerprint fp = new TestFingerprint(false);
+        fp.facets.setOwner(Saveable.NOOP);
+        TestFingperprintFacet facet = new TestFingperprintFacet(fp, System.currentTimeMillis(), false);
+        fp.facets.add(facet);
+        FingerprintCleanupThread cleanupThread = new TestFingerprintCleanupThread(fp);
+        cleanupThread.execute(testTaskListener);
+        assertFalse("Should have deleted obsolete file.", fpFile.toFile().exists());
```

Comment:
Hamcrest assertion for easier diagnosis of failures?
```suggestion
                assertThat(fpFile.toFile()., is(not(aReadableFile())));
        assertFalse("Should have deleted obsolete file.", fpFile.toFile().exists());
```

Reply:
@MarkEWaite IIUC the `aReadableFile ` FileMatcher is present inside Hamcrest Version 2.0 onwards. However we are using hamcrest-library version 1.3. Should I try upgrading it?

Reply:
should be fine to give it a go at upgrading, I don't remember hitting issues upgrading it elsewhere